internal/remote/crd.go

```diff
@@ -0,0 +1,37 @@
+package remote
+
+import (
+	"errors"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+func crdReady(crd *apiextensionsv1.CustomResourceDefinition) bool {
```

Comment:
There is another file named crd_upgrade. I think we can move these functions there since they all deal with the upgrade of the crd to the remote cluster

Reply:
I wanted to leave it separately on purpose, to make the scope of refactoring more visible.
It is going to be moved elsewhere anyway when implementing #2015.
For now I can move it into `crd_upgrade.go`, no problem.